### PR TITLE
Refactored generateRpcUrl out of protect button

### PR DESCRIPTION
**Pull request by odysseus0**
There is a need to display the RpcURL directly as text so the method that generates the URL needs to be exposed.

### CODE REVIEW

File: protect/src/button.tsx [hunk 1]
```diff
@@ -32,22 +32,20 @@ export interface ProtectButtonOptions extends PropsWithChildren {
...
+export const generateRpcUrl = ({
+  chainIdActual,
   hints,
   bundleId,
-  chainId,
-  children,
-  builders,
+  builders
+}: {
+  chainIdActual?: number;
```
Comment (zeroXbrock) [inline protect/src/button.tsx]: let's just call this `chainId` -- don't need to differentiate "actual" in this context

File: protect/src/button.tsx [hunk 2]
```diff
@@ -68,6 +66,23 @@ const FlashbotsProtectButton: FunctionComponent<ProtectButtonOptions> = ({
...
+const FlashbotsProtectButton: FunctionComponent<ProtectButtonOptions> = ({
+  addChain,
+  hints,
+  bundleId,
+  chainId,
+  children,
+  builders,
+}) => {
+  const chainIdActual: number = chainId || 1
+  const rpcUrl = generateRpcUrl(chainIdActual, hints, bundleId, builders)
```
Comment (zeroXbrock) [inline protect/src/button.tsx]: this throws a build error:
```
➜  protect git:(genRPCUrl) yarn build
yarn run v1.22.19
$ tsc
src/button.tsx:85:48 - error TS2554: Expected 1 arguments, but got 4.

85   const rpcUrl = generateRpcUrl(chainIdActual, hints, bundleId, builders)
                                                  ~~~~~~~~~~~~~~~~~~~~~~~~~


Found 1 error in src/button.tsx:85

error Command failed with exit code 2.
```

args need to be passed an an object for this function

File: protect/src/button.tsx (hunk 1 above)
Comment (odysseus0) [reply protect/src/button.tsx]: makes total sense!

File: protect/src/button.tsx (hunk 2 above)
Comment (odysseus0) [reply protect/src/button.tsx]: somehow forgot to include this line when making the commit... orz

Fixed!

**Comment by odysseus0** [comment]
Fixed based on feedback. Thanks a ton for the check!

**Comment by zeroXbrock** [comment]
nice work, thanks @odysseus0

**Comment by zeroXbrock** [comment]
merged in my own branch w/ some bugfixes of my own

**Pull request merged**